samples/boards/nrf52/power_mgr/src/main.c

```diff
@@ -0,0 +1,315 @@
+static void suspend_devices(void);
+/* Resume Devices */
+static void resume_devices(void);
+static struct device *device_list;
+static int device_count;
+static int post_ops_done = 1;
+
+/*
+ * Example ordered list to store devices on which
+ * device power policies would be executed.
+ */
+#define DEVICE_POLICY_MAX 15
+static char device_ordered_list[DEVICE_POLICY_MAX];
+static char device_retval[DEVICE_POLICY_MAX];
+int pm_state;
+
+/* change this to use another GPIO port */
+#define PORT    SW0_GPIO_NAME
+#define PIN	SW0_GPIO_PIN
+
+#define DEMO_DESCRIPTION	\
+	"Demo Description\n"	\
+	"Application creates Idleness, Due to which System Idle Thread is\n"\
+	"scheduled and it enters into various Low Power States.\n"\
+	"Press Button 1 on Board to wake device from Low Power State.\n"
+
+extern int nrf_gpiote_interrupt_enable(uint32_t mask);
+
+static void gpio_setup(void)
+{
+	struct device *gpiob;
+
+	gpiob = device_get_binding(PORT);
+	gpio_pin_configure(gpiob, PIN, GPIO_DIR_IN
+				| GPIO_PUD_PULL_UP
+				| GPIO_CFG_SENSE_LOW);
+	/* Enable GPIOTE Port Event */
+	nrf_gpiote_interrupt_enable(GPIOTE_INTENSET_PORT_Msk);
+	NVIC_EnableIRQ(GPIOTE_IRQn);
+}
+
+/* Print Demo Discription String */
+static void display_demo_description(void)
+{
+	printk(DEMO_DESCRIPTION);
+}
+
+/* Application main Thread */
+void main(void)
+{
+	printk("\n\n***Power Management Demo on %s****\n", CONFIG_ARCH);
+	display_demo_description();
+
+	/* Setup GPIO */
+	gpio_setup();
+	create_device_list();
+
+	while (1) {
+		printk("\nApplication main thread\n");
+		k_sleep(SECONDS_TO_SLEEP * 1000);
+	}
+}
+
+/* This Function decides which Low Power Mode to Enter based on Idleness.
+ * In actual Application this decision can be made using time (ticks)
+ * And RTC timer can be programmed to wake the device after time elapsed.
+ */
+static int check_pm_policy(s32_t ticks)
+{
+	static int policy = NO_POWER_SAVING_STATE;
+	int power_states[] = {SYS_POWER_STATE_MAX, SYS_POWER_STATE_CPU_LPS,
+			SYS_POWER_STATE_CPU_LPS_1, SYS_POWER_STATE_DEEP_SLEEP};
+	/*
+	 * Compare time available with wake latencies and select
+	 * appropriate power saving policy
+	 * For the demo we will alternate between following policies
+	 * 0 = no power saving operation
+	 * 1 = low power state Constant Latency
+	 * 2 = Low Power State
+	 * 3 = System Off State - Device will reset on Exit from this State
+	 */
+	policy = ++policy > MAX_POWER_SAVING_STATE ?
+					NO_POWER_SAVING_STATE : policy;
+	/* Pick Platform specific State */
+	return power_states[policy];
+}
+
+static void low_power_state_exit(void)
+{
+	/* Perform some Application specific task on Low Power Mode Exit */
+
+	/* Turn on suspended peripherals/clocks as necessary */
+	printk("---- Low power state exit !\n");
+}
+
+static void deep_sleep_exit(void)
+{
+	/* Turn on peripherals and restore device states as necessary */
+	resume_devices();
+	printk("====Deep sleep exit!\n");
+}
+
+static int low_power_state_entry(int pm_state)
```

Comment:
This parameters overlays the same named global variable. And it seems to be unneeded.

Reply:
I kept it same while keeping uniformity with other power management sample application .(samples/boards/nrf52/power_mgr/src/main.c)

I got your point, Sure I will modify it.